oak/server/rust/oak_runtime/src/runtime/mod.rs

```diff
@@ -679,4 +677,112 @@ impl Runtime {
 
         Ok(())
     }
+
+    pub fn new_runtime_proxy(self: Arc<Self>) -> RuntimeProxy {
+        RuntimeProxy {
+            runtime: self.clone(),
+            node_id: self.new_node_reference(),
+        }
+    }
+}
+
+/// A proxy object that binds together a reference to the underlying [`Runtime`] with a single
+/// [`NodeId`].
+///
+/// This can be considered as the interface to the [`Runtime`] that node and pseudo-node
+/// implementations have access to.
+///
+/// Individual methods simply forward to corresponding methods on the underlying [`Runtime`], by
+/// partially applying the first argument.
+#[derive(Clone)]
+pub struct RuntimeProxy {
```

Comment:
Are there any use cased when a pseudo-node will need to communicate with multiple nodes?
Or we can just create a new instance of the pseudo-node for each node that requires it?

Reply:
a `RuntimeProxy` instance is used by a single node instance to communicate with the runtime. I clarified a bit in the comment, does that make sense?

Reply:
Thanks